addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/handler/SqueezeBoxNotificationPlayer.java

```diff
@@ -0,0 +1,261 @@
+
+        try {
+            waitForPlaylistUpdate(listener);
+        } finally {
+            squeezeBoxServerHandler.unregisterSqueezeBoxPlayerListener(listener);
+        }
+    }
+
+    /**
+     * Monitor the number of playlist entries. When it changes, then we know the playlist
+     * has been updated with the notification URL. There's probably an edge case here where
+     * someone is updating the playlist at the same time, but that should be rare.
+     *
+     * @param listener
+     * @throws InterruptedException
+     * @throws SqueezeBoxTimeoutException
+     */
+    private void waitForPlaylistUpdate(SqueezeBoxNotificationListener listener)
+            throws InterruptedException, SqueezeBoxTimeoutException {
+        logger.trace("Waiting up to {} ms for playlist to be updated...", playlistCommandTimeout * 1000);
+
+        int timeoutCount = 0;
+
+        while (!listener.isPlaylistUpdated()) {
+            Thread.sleep(100);
+            if (timeoutCount++ > playlistCommandTimeout * 10) {
+                throw new SqueezeBoxTimeoutException("Unable to update playlist.");
+            }
+        }
+
+        this.playlistModified = true;
+    }
+
+    private void playNotification() throws InterruptedException, SqueezeBoxTimeoutException {
+        logger.debug("Playing notification");
+
+        notificationMessagePlaylistsIndex = squeezeBoxPlayerHandler.currentNumberPlaylistTracks() - 1;
+        SqueezeBoxNotificationListener listener = new SqueezeBoxNotificationListener(mac);
+        listener.resetStopped();
+
+        squeezeBoxServerHandler.registerSqueezeBoxPlayerListener(listener);
+        squeezeBoxServerHandler.playPlaylistItem(mac, notificationMessagePlaylistsIndex);
+
+        logger.trace("Waiting up to {} ms for stop...", notificationLengthTimeout * 1000);
+
+        try {
+            int timeoutCount = 0;
+
+            while (!listener.isStopped()) {
+                Thread.sleep(100);
+                if (timeoutCount++ > notificationLengthTimeout * 10) {
+                    throw new SqueezeBoxTimeoutException("Unable to play message.");
+                }
+            }
+        } finally {
+            squeezeBoxServerHandler.unregisterSqueezeBoxPlayerListener(listener);
+        }
+    }
+
+    private void restorePlayerState() {
+        logger.debug("Restoring player state");
+
+        // Resume playing save playlist item if player wasn't stopped.
+        // Note that setting the time doesn't work for remote streams.
+        if (!playerState.isStopped()) {
+            logger.debug("Resuming last item playing");
+            squeezeBoxServerHandler.playPlaylistItem(mac, playerState.getPlaylistIndex());
+            squeezeBoxServerHandler.setPlayingTime(mac, playerState.getPlayingTime());
+        } else if (!playerState.isPlaying()) {
+            logger.debug("Pausing the player");
+            squeezeBoxServerHandler.pause(mac);
```

Comment:
I guess we'll see during testing whether the player pauses reliably.  :-)

Reply:
During my first tests the pause worked ok; but with dynamic playlists we do not properly detect the end of a message. Probably we´ll have to add handling for a new song/track change to make this work.

Reply:
The dynamic playlist thing is more complex than expected; probably something for another PR.

Reply:
I agree.